rest_docs/rest-catalog-open-api-v0.1.yaml

```diff
@@ -0,0 +1,657 @@
+        in: path
+        required: true
+        schema:
+          type: string
+        examples:
+          singlepart_namespace:
+            value: "accounting"
+          multipart_namespace:
+            value: "accounting.tax"
+    get:
+      tags:
+        - Catalog API
+      summary: List all table identifiers underneath a given namespace
+      description: Return all table identifiers under this namespace
+      operationId: listTables
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ListTablesResponse'
+        '404':
+          description: Not Found (NoSuchNamespaceException)
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/responses/IcebergResponseObject'
+              example:
+                '{ error: { message: "Namespace does not exist", type: "NoSuchNamespaceException", code: 404 }'
+
+  /namespaces/{namespace}/tables/{table}:
+    parameters:
+      - name: namespace
+        in: path
+        description: A namespace identifier
+        required: true
+        schema:
+          type: string
+        examples:
+          singlepart_namespace:
+            value: "accounting"
+          multipart_namespace:
+            value: "accounting.tax"
+      - name: table
+        in: path
+        description: A table name
+        required: true
+        schema:
+          type: string
+        example: "sales"
+
+    delete:
+      tags:
+        - Catalog API
+      summary: Drop a table from the catalog
+      operationId: dropTable
+      description: Remove a table from the catalog
+      parameters:
+        - name: purgeRequested
+          in: query
+          required: false
+          description: Whether the user requested to purge the underlying table's data and metadata
+          schema:
+            type: boolean
+            default: false
+      responses:
+        200:
+          description: OK
+          content:
+            'application/json':
+              schema:
+                $ref: '#/components/schemas/DropTableResponse'
+              example: { "data": { "dropped": true, "purged": false } }
+        202:
+          # TODO - Add a callback or handle to check on the state of an async purge request
+          description: Accepted - for use if purgeRequested is implemented as an asynchronous action.
+        404:
+          $ref: '#/components/responses/NoSuchTableResponse'
+
+    head:
+      tags:
+        - Catalog API
+      summary: Check if a table exists
+      operationId: tableExists
+      description:
+        Check if a table exists within a given namespace.
+      responses:
+        200:
+          description: OK - Table Exists
+        401:
+          description: Unauthorized
+        404:
+          description: Not Found
+
+  /tables/rename:
```

Comment:
This was discussed (and I had it as that at one point).

Several people felt it was somewhat of an overloading of the API and would require people to look up the documentation of what some of the verbs do (i.e. what does PUT on table mean, without having to look it up). There was also concern that many backend frameworks can't handle a request body that might have multiple forms.

At one point I did have this implemented as a method of the original table, but that is one school of thought. 
I personally don't want to utilize REST semantics for purity sake, over making a useful API for people. I don't mean that using `PUT /namespaces/{ns}/table/{tbl}` with the new identifier would cause that, but I am advocating for balancing practicality with adhering to standards.

That was the discussion that got moved to `/dev/null` by Github due to a lot of conversation so I thought I'd summarize.  🙂